src/client/Components/Util/OAuth.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import Button from '@material-ui/core/Button';
+import PropTypes from 'prop-types';
+
+export default class OAuth extends React.Component {
+  state = {
+    user: {},
+    disabled: ''
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.startAuth = this.startAuth.bind(this);
+    this.closeCard = this.closeCard.bind(this);
+  }
+
+  componentDidMount() {
+    const { socket, provider } = this.props;
+
+    socket.on(provider, (user) => {
+      this.popup.close();
+      this.setState({ user });
+    });
+  }
+
+  checkPopup() {
+    const check = setInterval(() => {
+      const { popup } = this;
+      if (!popup || popup.closed || popup.closed === undefined) {
+        clearInterval(check);
+        this.setState({ disabled: '' });
+      }
+    }, 1000);
+  }
+
+  openPopup() {
+    const { provider, socket } = this.props;
+    const width = 600;
+    const height = 600;
+    const left = (window.innerWidth / 2) - (width / 2);
+    const top = (window.innerHeight / 2) - (height / 2);
+    const url = `http://v9_website.ngrok.io/api/auth/${provider}?socketId=${socket.id}`;
+
+    return window.open(url, '',
+      `toolbar=no, location=no, directories=no, status=no, menubar=no, 
+      scrollbars=no, resizable=no, copyhistory=no, width=${width}, 
+      height=${height}, top=${top}, left=${left}`);
+  }
+
+  startAuth(e) {
+    if (!this.state.disabled) {
+      e.preventDefault();
+      this.popup = this.openPopup();
+      this.checkPopup();
+      this.setState({ disabled: 'disabled' });
```

Comment:
Why are we using a string and not a boolean?

Reply:
Because of the tutorial I followed